joern-cli/frontends/jssrc2cpg/src/test/scala/io/joern/jssrc2cpg/passes/ast/SimpleAstCreationPassTest.scala

```diff
@@ -515,6 +513,44 @@ class SimpleAstCreationPassTest extends AbstractPassTest {
       accessElement.canonicalName shouldBe "foo"
     }
 
+    "be correct for call on object with argument" in AstFixture("""
+        |function method(x) {
+        |  a.b(x);
+        |}
+        """.stripMargin) { cpg =>
+      val List(method) = cpg.method.nameExact("method").l
+      val List(block)  = method.astChildren.isBlock.l
+
+      val List(call) = block.astChildren.isCall.l
+      call.code shouldBe "a.b(x)"
+      call.name shouldBe ""
+      call.dispatchType shouldBe DispatchTypes.DYNAMIC_DISPATCH
+
+      val List(receiver) = call.receiver.isCall.l
+      receiver.code shouldBe "a.b"
+      receiver.methodFullName shouldBe Operators.fieldAccess
+
+      val List(receiverAsArg) = call.argument(0).cast[Call].l
+      receiver shouldBe receiverAsArg
+
+      val List(base) = receiver.astChildren.isIdentifier.argumentIndex(1).l
+      base.name shouldBe "a"
```

Comment:
I don't think that's we want yet.

We want

- either the `a` identifier as arg 0 - apparently that's how it's always been with js2cpg
- or the `a.b` field access as arg 0 - that matches the original code more precisely, but I don't know if there was a reason why we've gone with the identifier previously
- but neither can be arg 1. That must be reserved for the "x" identifier that's actually in position 1 in the original code.

Reply:
Look closely. Thats the base of the receiver (a field access call here).

Reply:
oh, I see :facepalm: